yt_dlp/extractor/youtube.py

```diff
@@ -3702,17 +3702,16 @@ def _invalid_player_response(self, pr, video_id):
             return pr_id
 
     def _extract_player_responses(self, clients, video_id, webpage, master_ytcfg, smuggled_data):
-        initial_pr = None
+        initial_pr = ignore_initial_response = None
         if webpage:
             experiments = traverse_obj(master_ytcfg, (
-                'WEB_PLAYER_CONTEXT_CONFIGS', ..., 'serializedExperimentIds', {str}, {lambda x: x.split(',')}, ..., {str}))
-            if all(x in experiments for x in self._POTOKEN_EXPERIMENTS):
+                'WEB_PLAYER_CONTEXT_CONFIGS', ..., 'serializedExperimentIds', {lambda x: x.split(',')}, ...))
+            if 'web' in clients and all(x in experiments for x in self._POTOKEN_EXPERIMENTS):
```

Comment:
Maybe we want something like this?
```py
            if 'web' in clients:
                experiments = traverse_obj(master_ytcfg, (
                    'WEB_PLAYER_CONTEXT_CONFIGS', ..., 'serializedExperimentIds', {str}, {lambda x: x.split(',')}, ..., {str}))
                if all(x in experiments for x in self._POTOKEN_EXPERIMENTS):
```
that way we do experiments only if needed...